controller/web/registration.go

```diff
@@ -3,24 +3,95 @@ package web
 import (
 	"net/http"
 
+	"github.com/go-soa/auth/controller/web/request"
+	"github.com/go-soa/auth/model"
+	"github.com/go-soa/auth/repository"
+	"golang.org/x/crypto/bcrypt"
 	"golang.org/x/net/context"
 )
 
 // RegistrationIndex ...
 func (h *Handler) RegistrationIndex(ctx context.Context, rw http.ResponseWriter, r *http.Request) {
-	err := h.Tmpl.ExecuteTemplate(rw, "registration_index", nil)
+	err := h.Tmpl.ExecuteTemplate(rw, h.TmplName, nil)
 	if err != nil {
 		http.Error(rw, err.Error(), http.StatusInternalServerError)
 		return
 	}
+}
+
+// RegistrationCreate ...
+func (h *Handler) RegistrationCreate(ctx context.Context, rw http.ResponseWriter, r *http.Request) {
+	r.ParseForm()
 
-	formMail := "kamil.biela@gmail.com"
-	formUsername := "test"
+	registrationRequest := request.NewRegistrationRequestFromForm(r.Form)
+	validationErrorBuilder := registrationRequest.Validate()
```

Comment:
Hmm.. validation error builder shouldbe part of the context, and should be passed as an argument into `Validate()` method.